Assets/Src/Scripts/AgileMvvm/ViewModel.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace AgileMvvm {
+  // The ViewModel class that maintains models and raises model update events to the bound Views. It
+  // also provides the utility methods to manage the binding relationships.
+  //
+  // In AgileMvvm, the relationships among View, ViewModel and Model are:
```

Comment:
From the examples below, there isn't real model class. It's just properties in the ViewModel. Just want to clarify what's the definition of model. Is it the properties of the ViewModel, or the underlying SeedLang components?

Reply:
Per the definition from https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel, ViewModel exposes bindable properties to access the underlying Model instances. Thus, "The view model has been described as a state of the data in the model."

Many MVVM frameworks follow this definition. For example, `Microsoft.Toolkit.Mvvm` defines similar bindable properties in the ViewModel class, where the ViewModel class can then operate on the underlying Model class (such as a Database, a File, or an in-memory structure) to access the underlying data.

As for our SeedCalc app, the ViewModel defines the bindable properties to communicate with the View. Meanwhile, the ViewModel still needs to maintain a SeedLang executor to do the underlying calculation. Here, the SeedLang executor is the Model in the MVVM pattern.

I have updated the sample code inside the comments, to clarify that there could be an underlying instance that plays the Model role.